lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1116,7 +1129,16 @@ ChangeResult TypeAnalyzer::visitAtenFlattenUsingIntsOp(
       knowledge.hasSizes = true;
       for (auto i = 0; i < startDim; i++)
         knowledge.sizes.push_back(operand.sizes[i]);
-      knowledge.sizes.push_back(kUnknownSize);
+      int64_t dimProduct = 1;
+      bool hasDynamicDims = false;
+      for (auto i = startDim; i <= endDim; i++) {
+        if (operand.sizes[i] == kUnknownSize) {
+          hasDynamicDims = true;
```

Comment:
nit: could let `dimProduct = kUnknownSize ` here, then `hasDynamicDims ` can be removed.

Reply:
Good point. I didn't look up what kUnknownSize was.